wv2util/ExperimentalFeature.cs

```diff
@@ -0,0 +1,112 @@
+        public string Name { get; set; }
+        
+        public bool isEnabled;
+
+        public bool IsEnabled
+        {
+            get { return isEnabled; }
+            set
+            {
+                if (value)
+                {
+                    turnOn();
+                }
+                else
+                {
+                    turnOff();
+                }
+                isEnabled = value;
+            }
+        }
+
+        public string Description { get; set; }
+
+        private Action turnOn, turnOff;
+
+        public ExperimentalFeature(Action turnOn, Action turnOff, Func<bool> isOn)
+        {
+            this.turnOn = turnOn;
+            this.turnOff = turnOff;
+            isEnabled = isOn();
+        }
+
+        static public bool IsEnabledForApp(string commandLine, string envVars)
+        {
+            // TODO implement
+            return false;
+        }
+
+        public int CompareTo(ExperimentalFeature other)
+        {
+            return Name.CompareTo(other.Name);
+        }
+
+        public bool Equals(ExperimentalFeature other)
+        {
+            return Name.Equals(other.Name);
+        }
+    }
+
+    public class EnvVarExperimentalFeature : ExperimentalFeature
+    {
+        public EnvVarExperimentalFeature(string envVar, string onVal, string offVal) : base(
+            () =>
+            {
+                // Turn on:
+                Environment.SetEnvironmentVariable(envVar, onVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                // Turn off:
+                Environment.SetEnvironmentVariable(envVar, offVal, EnvironmentVariableTarget.User);
+            },
+            () =>
+            {
+                string val = Environment.GetEnvironmentVariable(envVar, EnvironmentVariableTarget.User);
+                return val != null && val == onVal;
+            })
+        {
+        }
+    }
+
+    public class ExperimentalFeatureList : ObservableCollection<ExperimentalFeature>
+    {
+        private List<ExperimentalFeature> m_List = new List<ExperimentalFeature>();
+
+        public ExperimentalFeatureList()
+        {
+            // Visual Hosting
+            Items.Add(new EnvVarExperimentalFeature(
+                "COREWEBVIEW2_FORCED_HOSTING_MODE",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_VISUAL",
+                "COREWEBVIEW2_HOSTING_MODE_WINDOW_TO_WINDOW")
+            {
+                Name = "Visual hosting",
+                Description = "When on apps will be running in Visual Hosting instead of regular Window hosting"
+            });
+
+            // Canary self-hosting
+            Items.Add(new EnvVarExperimentalFeature(
+                "WEBVIEW2_RELEASE_CHANNEL_PREFERENCE",
+                "1",
+                null)
+            {
+                Name = "Canary self-hosting",
+                Description = "When on apps will use canary WebView2 runtime if installed instead of stable"
```

Comment:
Checking this check box should check if canary is installed and if not, prompt the end user and then start the install. This will also be useful for the overrides tab when checking different branches. Or also seems good if you don't want to do that in this PR.

Reply:
How do I check if canary is installed programmatically?

Reply:
We can use the AppState.RuntimesList which has list of installed runtimes

Reply:
done!